.github/workflows/ci_tests.yml

```
@@ -103,6 +103,14 @@ jobs:
         OPTIMADE_DATABASE_BACKEND: mongodb
       run: pytest -v --cov=./aiida_optimade/ --cov-report=xml:mongo_cov.xml --durations=20
 
+    - name: Test with pytest (Data Curation)
+      env:
+        AIIDA_TEST_BACKEND: ${{ matrix.backend }}
+        AIIDA_TEST_PROFILE: test_${{ matrix.backend }}
+        AIIDA_PROFILE: test_${{ matrix.backend }}
+        PYTEST_OPTIMADE_CONFIG_FILE: ./tests/static/test_data_curation_config.json
+      run: pytest -v --cov=./aiida_optimade/ --cov-report=xml --durations=20 tests/server/routers/test_structures.py::test_structures_endpoint_data
```

Ensure the current coverage report is not overwritten (the one created in the "Test with pytest (AiiDA)" step).
```suggestion
      run: pytest -v --cov=./aiida_optimade/ --cov-report=xml --cov-append --durations=20 tests/server/routers/test_structures.py::test_structures_endpoint_data
```

---

Added, but somehow the coverage still decreases.

---

Could you test commenting out this step in the job and see if the coverage changes? I might be good to understand what the issue is. This _should_ work, meaning the coverage found through this step should be added to the existing `coverage.xml` file. It might however work if the coverage file name is added explicitly? I.e., changing the `--cov-report` value from `xml` to `xml:coverage.xml` for both this step and the step above with the first `pytest` run.

---

Sorry, I just saw this, I'll test this and request your review then.

---

Did you look into this?

---

Move the test right after the AiiDA test with `--coz-append` works.